scripts/pathseq/wdl/pathseq_pipeline.wdl

```diff
@@ -20,179 +26,712 @@
+  Int? filter_bwa_seed_length
+  Int? min_clipped_read_length
+  Int? max_masked_bases
+  Int? min_base_quality
+  Int? quality_threshold
+  Int? dust_mask_quality
+  Int? dust_window
+  Float? dust_t
+  Int? max_adapter_mistmatches
+  Int? min_adapter_length
+  Int? filter_reads_per_partition
+  Int? filter_bam_partition_size
   Boolean? skip_pre_bwa_repartition
+
+  # Alignment options
+  Int? microbe_min_seed_length
+  Int? max_alternate_hits
+  Int? bwa_score_threshold
+
+  # Taxonomy scoring options
   Boolean? divide_by_genome_length
-  Int? filter_bwa_seed_length
+  Float? min_score_identity
+  Float? identity_margin
+  Boolean? not_normalized_by_kingdom
+  Int? score_reads_per_partition_estimate
+
+  # Runtime parameters
+  String gatk_docker
+  String samtools_docker = "biocontainers/samtools:v1.9-4-deb_cv1"
+  String genomes_in_the_cloud_docker = "broadinstitute/genomes-in-the-cloud:2.3.1-1512499786"
+  String linux_docker = "ubuntu:18.10"
+
+  File? gatk4_jar_override
+
+  Int? cram_to_bam_preemptible_attempts
+  Int? downsample_preemptible_attempts
+  Int? filter_preemptible_attempts
+  Int? align_preemptible_attempts
+  Int? score_preemptible_attempts
+  Int? process_filter_metrics_preemptible_attempts
+  Int? process_score_metrics_preemptible_attempts
+
+  Int? cram_to_bam_cpu
+  Int? filter_cpu
+  Int? align_cpu
+  Int? score_cpu
+
+  Float? cram_to_bam_mem_gb
+  Int? filter_mem_gb
+  Int? align_mem_gb
+  Int? score_mem_gb
+
+  Boolean? filter_ssd
+  Boolean? align_ssd
+  Boolean? score_ssd
+
+  Int? cram_to_bam_max_retries
+
+  # Optional input to increase all disk sizes in case of outlier sample with strange size behavior
+  Int? cram_to_bam_additional_disk_gb
+  Int? downsample_additional_disk_gb
+  Int? filter_additional_disk_gb
+  Int? align_additional_disk_gb
+  Int? score_additional_disk_gb
+
+  Boolean is_bam = basename(input_bam_or_cram, ".bam") + ".bam" == basename(input_bam_or_cram)
+
+  # Convert to BAM if we have a CRAM
+  if (!is_bam) {
+    call CramToBam {
+      input:
+        cram_file = input_bam_or_cram,
+        cram_index = select_first([input_bam_or_cram_index]),
+        reference_fasta = select_first([cram_reference_fasta]),
+        reference_index = select_first([cram_reference_fasta_index]),
+        docker = samtools_docker,
+        cpu = cram_to_bam_cpu,
+        mem_gb = cram_to_bam_mem_gb,
+        extra_disk_gb = cram_to_bam_additional_disk_gb,
+        preemptible = cram_to_bam_preemptible_attempts,
+        max_retries = cram_to_bam_max_retries,
+    }
+  }
+
+  File bam_file = select_first([CramToBam.bam_file, input_bam_or_cram])
+  File? bam_index = if defined(CramToBam.bam_index) then CramToBam.bam_index else input_bam_or_cram_index
+
+  # Downsample bam for filter metrics estimation
+  if (downsample) {
+    call Downsample {
+      input:
+        input_bam_file=bam_file,
+        input_bam_index_file=bam_index,
+        downsampled_bam_filename="${sample_name}.downsampled.bam",
```

Comment:
While you're here, it would be great to rev this to WDL 1.0.

Reply:
Done

Reply:
Done